cmd/xl-v1-metadata.go

```diff
@@ -17,20 +17,61 @@
 package cmd
 
 import (
+	"bytes"
+	"encoding/hex"
 	"encoding/json"
 	"errors"
 	"path"
-	"runtime"
 	"sort"
 	"sync"
 	"time"
+
+	"github.com/aead/poly1305"
+	"github.com/minio/minio/pkg/bitrot"
+	sha256 "github.com/minio/sha256-simd"
+	"golang.org/x/crypto/blake2b"
 )
 
 const (
 	// Erasure related constants.
 	erasureAlgorithmKlauspost = "klauspost/reedsolomon/vandermonde"
 )
 
+// DefaultBitrotAlgorithm is the default algorithm used for bitrot protection
+const DefaultBitrotAlgorithm = bitrot.Poly1305
```

Comment:
In general yes - on arm64 github.com/aead/poly1305 will use the reference implementation. If this is slower than SHA-256, we can implement it in assembly: See [sum_arm.s](https://github.com/golang/crypto/blob/master/poly1305/sum_arm.s)